tests/file_writer/test_recording.py

```diff
@@ -1724,3 +1660,96 @@ def test_FileWriterProcess__deletes_recorded_stim_data_after_stop_time(
         )
 
         this_file.close()
+
+
+def test_FileWriterProcess__upload_thread_gets_added_to_container_after_all_files_get_finalized(
+    four_board_file_writer_process, mocker
+):
+    update_customer_settings_command = copy.deepcopy(GENERIC_UPDATE_CUSTOMER_SETTINGS)
+    update_customer_settings_command["config_settings"]["auto_delete_local_files"] = False
+    update_customer_settings_command["config_settings"]["auto_upload_on_completion"] = True
+    file_writer_process_ready_for_upload = file_writer_process_with_closed_h5_files_for_upload(
+        four_board_file_writer_process=four_board_file_writer_process,
+        update_customer_settings_command=update_customer_settings_command,
+    )
+    file_writer_process = file_writer_process_ready_for_upload["fw_process"]
+    to_main_queue = file_writer_process_ready_for_upload["to_main_queue"]
+    mocker.patch.object(file_uploader.ErrorCatchingThread, "errors", autospe=True, return_value=True)
+    mocker.patch.object(
+        file_uploader.ErrorCatchingThread, "get_error", autospe=True, return_value="mocked_error"
+    )
+
+    invoke_process_run_and_check_errors(file_writer_process, num_iterations=6)
+    # pylint: disable=protected-access
+    assert (
+        file_writer_process._customer_settings == GENERIC_UPDATE_CUSTOMER_SETTINGS["config_settings"]
+    )  # pylint: disable=protected-access
+    assert (
+        file_writer_process._is_finalizing_files_after_recording() is False
+    )  # pylint: disable=protected-access
+    assert len(file_writer_process._open_files[0].keys()) == 0  # pylint: disable=protected-access
+    assert len(file_writer_process._upload_threads_container) == 1  # pylint: disable=protected-access
+    assert (
+        file_writer_process._upload_threads_container[0]["failed_upload"] is False
+    )  # pylint: disable=protected-access
```

Comment:
It looks like a lot of assertions are being made on this object, I would just make a getter for it in file_writer to avoid having the messy `# pylint: disable=protected-access` everywhere. One other thing is pylint disables need to actually be directly on the line where the thing that they're disabling is happening. These aren't actually doing anything here, I believe it's not causing any pylint errors though because of the protected-access disable on line 1683 which is on its own line